### PR TITLE
James 2479 : json related libs upgrade

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

**Comment by trantienduchn** [comment]
```
ERROR: Error fetching remote repo 'origin'
hudson.plugins.git.GitException: Failed to fetch from https://github.com/linagora/james-jenkins
	at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:735)
	at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:983)
	at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1016)
	at org.jenkinsci.plugins.workflow.steps.scm.SCMStep.checkout(SCMStep.java:109)
	at org.jenkinsci.plugins.workflow.cps.CpsScmFlowDefinition.create(CpsScmFlowDefinition.java:109)
	at org.jenkinsci.plugins.workflow.cps.CpsScmFlowDefinition.create(CpsScmFlowDefinition.java:58)
	at org.jenkinsci.plugins.workflow.job.WorkflowRun.run(WorkflowRun.java:179)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
```
We usually get this error in recent days

**Comment by chibenwa** [comment]
test this please

Looks like bad connectivity with github

**Comment by rouazana** [comment]
```
[760a68095cb6d8f107480a75a0f48c383838fe01] Tests in error: 
[760a68095cb6d8f107480a75a0f48c383838fe01]   TikaTextExtractorTest.asListOfStringShouldReturnAListWhenMultipleElements:232 » NullPointer
```

File: pom.xml [hunk 1]
```diff
@@ -1905,8 +1905,8 @@
             </dependency>
             <dependency>
                 <groupId>net.javacrumbs.json-unit</groupId>
-                <artifactId>json-unit-fluent</artifactId>
-                <version>1.5.5</version>
+                <artifactId>json-unit-assertj</artifactId>
+                <version>2.0.0.RC1</version>
```
Comment (rouazana) [inline pom.xml]: not a final version?

**Comment by chibenwa** [comment]
test this please

File: pom.xml (hunk 1 above)
Comment (mbaechler) [reply pom.xml]: yes, we spoke about that during our meeting about lib upgrades and my comment about why was written in the doc : there's no stable release of json-unit-assertj yet but as it allows to drop non-standard fluent API, I proposed to use it anyway.

**Comment by mbaechler** [comment]
```
[a2a24341ffb2ff2b2a8dbcbdfa510d77862906a8] 
[a2a24341ffb2ff2b2a8dbcbdfa510d77862906a8] messageWithDotsInHeaderShouldBeIndexed  Time elapsed: 0.618 sec  <<< ERROR!
[a2a24341ffb2ff2b2a8dbcbdfa510d77862906a8] java.lang.NoSuchMethodError: com.fasterxml.jackson.dataformat.smile.SmileGenerator.getOutputContext()Lcom/fasterxml/jackson/core/json/JsonWriteContext;
[a2a24341ffb2ff2b2a8dbcbdfa510d77862906a8] 	at org.apache.james.mailbox.store.search.AbstractMessageSearchIndexTest.messageWithDotsInHeaderShouldBeIndexed(AbstractMessageSearchIndexTest.java:416)
[a2a24341ffb2ff2b2a8dbcbdfa510d77862906a8] 
```

**Comment by mbaechler** [comment]
test this please

**Comment by chibenwa** [comment]
test this please (build output lost)

**Comment by chibenwa** [comment]
test this please (build result lost)

**Comment by chibenwa** [comment]
```
[29f198d46fdf8d922a4368f010c8224722a9819b] Failed tests: 
[29f198d46fdf8d922a4368f010c8224722a9819b]   ErrorRoutesTest.defineJsonExtractExceptionShouldReturnBadRequestJsonFormat:98 1 expectation failed.
[29f198d46fdf8d922a4368f010c8224722a9819b] JSON path details doesn't match.
[29f198d46fdf8d922a4368f010c8224722a9819b] Expected: Unrecognized token 'a': was expecting ('true', 'false' or 'null')\n at [Source: a non valid JSON; line: 1, column: 2]
[29f198d46fdf8d922a4368f010c8224722a9819b]   Actual: Unrecognized token 'a': was expecting ('true', 'false' or 'null')
[29f198d46fdf8d922a4368f010c8224722a9819b]  at [Source: (String)"a non valid JSON"; line: 1, column: 2]
```

**Comment by chibenwa** [comment]
Forced pushed to solve conflict

**Comment by mbaechler** [comment]
test this please

**Comment by mbaechler** [comment]
test this please

**Comment by mbaechler** [comment]
problem triggering the CI, trying to open a new PR from this one

**Pull request closed without merging**